x-pack/plugins/fleet/public/components/agentless_enrollment_flyout/step_confirm_enrollment.tsx

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EuiButton, EuiPanel, EuiText, EuiLink, EuiSpacer, EuiCallOut } from '@elastic/eui';
+
+import type { Agent, AgentPolicy } from '../../types';
+import { useStartServices } from '../../hooks';
+import { AgentDetailsIntegrations } from '../../applications/fleet/sections/agents/agent_details_page/components/agent_details/agent_details_integrations';
+
+export const AgentlessStepConfirmEnrollment = ({
+  agent,
+  agentPolicy,
+  integrationTitle,
+}: {
+  agent?: Agent;
+  agentPolicy?: AgentPolicy;
+  integrationTitle: string;
+}) => {
+  const { docLinks } = useStartServices();
+  const [overallState, setOverallState] = useState<'pending' | 'success' | 'failure'>('pending');
+
+  // Calculate overall UI state from agent status
+  useEffect(() => {
+    if (agent && agent.status === 'online') {
+      setOverallState('success');
+    } else if (agent && (agent.status === 'error' || agent.status === 'degraded')) {
```

Comment:
Does `agent.status` look if there are errors with other `event.dataset`  such as `cloudbeat`  or `filebeat`. I see the status as Healthy; however, I see  Cloudbeat permission errors

Reply:
agent status should take into account all the agent sub-component states. the agent info here polls every 30 seconds, so perhaps there is a delay in the status getting updated here vs when the cloudbeat errors came in?